File: api/constants/enums.py
"""Centralized StrEnum definitions for the Autopolio backend.

StrEnum inherits from str, so enum values work seamlessly with string
comparisons, JSON serialization, and SQLAlchemy column defaults:
    JobStatus.PENDING == "pending"  # True
"""

from enum import StrEnum


class JobStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class LLMProvider(StrEnum):
    OPENAI = "openai"
    ANTHROPIC = "anthropic"
    GEMINI = "gemini"


class CLIType(StrEnum):
    CLAUDE_CODE = "claude_code"
    GEMINI_CLI = "gemini_cli"
    CODEX_CLI = "codex_cli"


class DocumentFormat(StrEnum):
    DOCX = "docx"
    PDF = "pdf"
    MD = "md"
    HTML = "html"


class ProjectType(StrEnum):
    PERSONAL = "personal"
    COMPANY = "company"
    OPEN_SOURCE = "open-source"


class ProjectStatus(StrEnum):
    COMPLETED = "completed"
    IN_PROGRESS = "in-progress"
    MAINTAINED = "maintained"


class EmploymentType(StrEnum):
    FULL_TIME = "full-time"
    CONTRACT = "contract"
    FREELANCE = "freelance"
    INTERN = "intern"


class GraduationStatus(StrEnum):
    GRADUATED = "graduated"
    ENROLLED = "enrolled"
    COMPLETED = "completed"
    WITHDRAWN = "withdrawn"


class ActivityType(StrEnum):
    VOLUNTEER = "volunteer"
    EXTERNAL = "external"


class PublicationType(StrEnum):
    JOURNAL = "journal"
    CONFERENCE = "conference"
    BOOK = "book"
    PATENT = "patent"


class SummaryStyle(StrEnum):
    PROFESSIONAL = "professional"
    CASUAL = "casual"
    TECHNICAL = "technical"

File: frontend/src/constants/enums.ts
/**
 * Centralized enum constants for the Autopolio frontend.
 * Uses `as const` pattern for type-safe string unions.
 */

// --- LLM Providers ---
export const LLM_PROVIDERS = {
  OPENAI: 'openai',
  ANTHROPIC: 'anthropic',
  GEMINI: 'gemini',
} as const

export type LLMProviderType = (typeof LLM_PROVIDERS)[keyof typeof LLM_PROVIDERS]

// --- CLI Types ---
export const CLI_TYPES = {
  CLAUDE_CODE: 'claude_code',
  GEMINI_CLI: 'gemini_cli',
  CODEX_CLI: 'codex_cli',
} as const

export type CLIType = (typeof CLI_TYPES)[keyof typeof CLI_TYPES]

// --- AI Modes ---
export const AI_MODES = {
  CLI: 'cli',
  API: 'api',
} as const

export type AIMode = (typeof AI_MODES)[keyof typeof AI_MODES]

// --- Job Status ---
export const JOB_STATUS = {
  PENDING: 'pending',
  RUNNING: 'running',
  COMPLETED: 'completed',
  FAILED: 'failed',
  CANCELLED: 'cancelled',
} as const

export type JobStatusType = (typeof JOB_STATUS)[keyof typeof JOB_STATUS]

// --- Document Formats ---
export const DOCUMENT_FORMATS = {
  DOCX: 'docx',
  PDF: 'pdf',
  MD: 'md',
  HTML: 'html',
} as const

export type DocumentFormatType = (typeof DOCUMENT_FORMATS)[keyof typeof DOCUMENT_FORMATS]
